lib/awful/mouse/init.lua

```diff
@@ -155,7 +155,10 @@ end
 --- Move a client.
 -- @param c The client to move, or the focused one if nil.
 -- @param snap The pixel to snap clients.
-function mouse.client.move(c, snap)
+-- @param finished An optional callback function, that will be called
+--                 when moving the client has been finished. The client
+--                 that has been moved will be passed to that function.
+function mouse.client.move(c, snap, finished)
```

Comment:
Any preferences on which of those three? (I would use `on_finish`, but will change it to whatever you decide.)